HW_3/app.py

```diff
@@ -0,0 +1,46 @@
+from flask import Flask, render_template, redirect, url_for, flash
+from flask_wtf import FlaskForm
+from wtforms import StringField, PasswordField, SubmitField
+from wtforms.validators import InputRequired, Email, Length
+from flask_sqlalchemy import SQLAlchemy
+from werkzeug.security import generate_password_hash, check_password_hash
+
+app = Flask(__name__)
+app.config['SECRET_KEY'] = 'HomeWork3'
+app.config['SQLALCHEMY_DATABASE_URI'] = 'sqlite:///users.db'  
+db = SQLAlchemy(app)
+
+class User(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    first_name = db.Column(db.String(50), nullable=False)
+    last_name = db.Column(db.String(50), nullable=False)
+    email = db.Column(db.String(120), unique=True, nullable=False)
+    password = db.Column(db.String(200), nullable=False)
+
+class RegistrationForm(FlaskForm):
+    first_name = StringField('Имя', validators=[InputRequired(), Length(min=2, max=50)])
+    last_name = StringField('Фамилия', validators=[InputRequired(), Length(min=2, max=50)])
+    email = StringField('Email', validators=[InputRequired(), Email(), Length(max=120)])
+    password = PasswordField('Пароль', validators=[InputRequired(), Length(min=6, max=200)])
+    submit = SubmitField('Зарегистрироваться')
+
+@app.route('/register', methods=['GET', 'POST'])
+def register():
+    form = RegistrationForm()
+
+    if form.validate_on_submit():
+        hashed_password = generate_password_hash(form.password.data, method='pbkdf2:sha256')
+        new_user = User(first_name=form.first_name.data, last_name=form.last_name.data, email=form.email.data, password=hashed_password)
+        db.session.add(new_user)
+        db.session.commit()
+        flash('Вы успешно зарегистрированы!', 'success')
+        return redirect(url_for('register'))
+
```

Comment:
Порой записи в БД не происходит и не понятно в чем ошибка. Все дело в том, что  форма не проходит валидацию и на этот счет не предусмотрены никакие действия
```suggestion
    else:
        for obj in form:
            for error in obj.errors:
                flash(error)
```